Dev4Agriculture.ISO11783.ISOXML/ISOXML.cs

```diff
@@ -273,6 +275,31 @@ public static ISOXML Load(string path, bool loadBinData = true)
             return isoxml;
         }
 
+        /// <summary>
+        /// Load an ISOXML TaskSet and return an ISOXML Object
+        /// </summary>
+        /// <param name="stream">zip file stream</param>
+        /// <param name="loadBinData">Shall all binary data such as grids and TLGs be loaded? Default is true</param>
+        /// <returns></returns>
+        public static ISOXML Load(Stream stream, bool loadBinData = true)
+        {
+            var path = Path.GetTempPath();
+            using (ZipArchive archive = new ZipArchive(stream, ZipArchiveMode.Read))
+            {
+                var fileNames = archive.Entries.Select(e => e.FullName).ToList();
+                if (!fileNames.Any(x => x.Contains("TASKDATA.XML", StringComparison.OrdinalIgnoreCase)))
+                {
+                    throw new InvalidDataException("Archive has incorrect data included!");
+                }
+
+                var folderName = fileNames.First().Substring(0, fileNames.First().IndexOf('/'));
+                archive.ExtractToDirectory(path, true);
+                path = Path.Combine(path, folderName);
+            }
+
+            return Load(path, loadBinData);
```

Comment:
At this point we exit the function without deleting the Data from the TEMP-Folder. Please add a step to remove the data from TEMP